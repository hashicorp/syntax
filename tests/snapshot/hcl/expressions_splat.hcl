attr = [for o in var.list : o.id]

attr = var.list[*].id

attr = var.list[*].interfaces[0].name

attr = var.list.*.id

attr = "${var.list.*.id}"

attr = "${var.list.*}"

attr = var.list.*
