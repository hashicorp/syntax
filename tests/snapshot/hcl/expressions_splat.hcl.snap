>attr = [for o in var.list : o.id]
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^ source.hcl punctuation.section.brackets.begin.hcl
#        ^^^ source.hcl keyword.control.hcl
#           ^ source.hcl
#            ^ source.hcl variable.other.readwrite.hcl
#             ^ source.hcl
#              ^^ source.hcl keyword.operator.word.hcl
#                ^ source.hcl
#                 ^^^ source.hcl variable.other.readwrite.hcl
#                    ^ source.hcl keyword.operator.accessor.hcl
#                     ^^^^ source.hcl variable.other.member.hcl
#                         ^ source.hcl
#                          ^ source.hcl keyword.operator.hcl
#                           ^ source.hcl
#                            ^ source.hcl variable.other.readwrite.hcl
#                             ^ source.hcl keyword.operator.accessor.hcl
#                              ^^ source.hcl variable.other.member.hcl
#                                ^ source.hcl punctuation.section.brackets.end.hcl
>
>attr = var.list[*].id
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^^^ source.hcl
#          ^ source.hcl keyword.operator.accessor.hcl
#           ^^^^ source.hcl variable.other.member.hcl
#               ^ source.hcl punctuation.section.brackets.begin.hcl
#                ^ source.hcl keyword.operator.splat.hcl
#                 ^ source.hcl punctuation.section.brackets.end.hcl
#                  ^ source.hcl keyword.operator.accessor.hcl
#                   ^^ source.hcl variable.other.member.hcl
>
>attr = var.list[*].interfaces[0].name
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^^^ source.hcl
#          ^ source.hcl keyword.operator.accessor.hcl
#           ^^^^ source.hcl variable.other.member.hcl
#               ^ source.hcl punctuation.section.brackets.begin.hcl
#                ^ source.hcl keyword.operator.splat.hcl
#                 ^ source.hcl punctuation.section.brackets.end.hcl
#                  ^ source.hcl keyword.operator.accessor.hcl
#                   ^^^^^^^^^^ source.hcl variable.other.member.hcl
#                             ^ source.hcl punctuation.section.brackets.begin.hcl
#                              ^ source.hcl constant.numeric.integer.hcl
#                               ^ source.hcl punctuation.section.brackets.end.hcl
#                                ^ source.hcl keyword.operator.accessor.hcl
#                                 ^^^^ source.hcl variable.other.member.hcl
>
>attr = var.list.*.id
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^^^ source.hcl
#          ^ source.hcl keyword.operator.accessor.hcl
#           ^^^^ source.hcl variable.other.member.hcl
#               ^ source.hcl keyword.operator.accessor.hcl
#                ^ source.hcl keyword.operator.splat.hcl
#                 ^ source.hcl keyword.operator.accessor.hcl
#                  ^^ source.hcl variable.other.member.hcl
>
>attr = "${var.list.*.id}"
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^ source.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#        ^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.other.interpolation.begin.hcl
#          ^^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl variable.other.readwrite.hcl
#             ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.accessor.hcl
#              ^^^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl variable.other.member.hcl
#                  ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.accessor.hcl
#                   ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.splat.hcl
#                    ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.accessor.hcl
#                     ^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl variable.other.member.hcl
#                       ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.other.interpolation.end.hcl
#                        ^ source.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>
>attr = "${var.list.*}"
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^ source.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#        ^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.other.interpolation.begin.hcl
#          ^^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl variable.other.readwrite.hcl
#             ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.accessor.hcl
#              ^^^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl variable.other.member.hcl
#                  ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.accessor.hcl
#                   ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.operator.splat.hcl
#                    ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.other.interpolation.end.hcl
#                     ^ source.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>
>attr = var.list.*
#^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#    ^ source.hcl variable.declaration.hcl
#     ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#      ^ source.hcl variable.declaration.hcl
#       ^^^ source.hcl
#          ^ source.hcl keyword.operator.accessor.hcl
#           ^^^^ source.hcl variable.other.member.hcl
#               ^ source.hcl keyword.operator.accessor.hcl
#                ^ source.hcl keyword.operator.splat.hcl
>